recipes-kernel/kernel-tests/kernel-test-nohz-files/nohz_test.c

```diff
@@ -363,41 +367,43 @@ static void validate_results(struct histogram_data *h)
 {
 	uint64_t p_99_999;
 	uint64_t p_99_9999;
+	const char *stats_summary = "Total count: %lu (samples)\n"
+                                    "Maximum latency: %lu (ns)\n"
+                                    "99.999 percentile: %lu (ns)\n"
+                                    "99.9999 percentile: %lu (ns)\n";
+	char buffer[512];
 
 	if (!h)
-		error_exit("Results validation failed; no histogram data");
+		error_exit("Results validation failed; no histogram data", NULL);
 
 	p_99_999 = get_percentile(99.999, h);
 	p_99_9999 = get_percentile(99.9999, h);
 
-	printf("Total count: %lu (samples)\n", h->cnt);
-	printf("Maximum latency: %lu (ns)\n", h->max);
-	printf("99.999 percentile: %lu (ns)\n", p_99_999);
-	printf("99.9999 percentile: %lu (ns)\n", p_99_9999);
+	sprintf(buffer, stats_summary, h->cnt, h->max, p_99_999, p_99_9999);
```

Comment:
Would it make sense to make `error_exit` variadic and use [`vprintf`][1] so that `buffer` and `sprintf` are unnecessary here?

If not, at least be sure to avoid overruns:
```suggestion
	snprintf(buffer, sizeof(buffer), stats_summary, h->cnt, h->max, p_99_999, p_99_9999);
```

[1]: <https://linux.die.net/man/3/printf>

Reply:
Good idea. Updated.